lib/matplotlib/tests/test_contour.py

```diff
@@ -478,3 +478,49 @@ def test_contour_autolabel_beyond_powerlimits():
     ax.clabel(cs)
     # Currently, the exponent is missing, but that may be fixed in the future.
     assert {text.get_text() for text in ax.texts} == {"0.25", "1.00", "4.00"}
+
+
+def test_countourf_legend_elements():
+    from matplotlib.patches import Rectangle
+    x = np.arange(1, 10)
+    y = x.reshape(-1, 1)
+    h = x * y
+
+    cs = plt.contourf(h, levels=[10, 30, 50],
+                      colors=['#FFFF00', '#FF00FF', '#00FFFF'],
+                      extend='both')
+    cs.cmap.set_over('red')
+    cs.cmap.set_under('blue')
+    cs.changed()
+    artists, labels = cs.legend_elements()
+    assert labels == ['$x \\leq -1e+250s$',
+                      '$10.0 < x \\leq 30.0$',
+                      '$30.0 < x \\leq 50.0$',
+                      '$x > 1e+250s$']
+    expected_colors = ((0.0, 0.0, 1.0, 1.0),
+                       (1.0, 1.0, 0.0, 1.0),
+                       (1.0, 0.0, 1.0, 1.0),
+                       (1.0, 0.0, 0.0, 1.0))
+    assert all(isinstance(a, Rectangle) for a in artists)
+    assert all(a.get_facecolor() == c for a, c in zip(artists,
+                                                      expected_colors))
```

Comment:
```suggestion
    assert all(mcolors.same_color(a.get_facecolor(), c)
               for a, c in zip(artists, expected_colors))
```
and you don't need to write `expected_colors` in its final form, but can use the original colour array you passed.

Reply:
Ahh, nice!